UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs

```diff
@@ -869,4 +871,140 @@ public void CmdRequestPaperEdit(GameObject paper, string newMsg)
 		}
 	}
 
+	/// <summary>
+	/// Performs a hug from one player to another.
+	/// </summary>
+	[Command]
+	public void CmdRequestHug(string hugger, GameObject huggedPlayer)
+	{
+		string huggee = huggedPlayer.GetComponent<PlayerScript>().playerName;
+		var huggedPlayerRegister = huggedPlayer.GetComponent<RegisterPlayer>();
+		ChatRelay.Instance.AddToChatLogServer(new ChatEvent
+		{
+			channels = ChatChannel.Local,
+			message = $"{hugger} has hugged {huggee}.",
+			position = huggedPlayerRegister.WorldPosition.To2Int()
+		});
+	}
+
+	/// <summary>
+	///	Performs a CPR action from one player to another.
+	/// </summary>
+	[Command]
+	public void CmdRequestCPR(GameObject rescuer, GameObject cardiacArrestPlayer)
+	{
+		var cardiacArrestPlayerRegister = cardiacArrestPlayer.GetComponent<RegisterPlayer>();
+
+		if (doingCPR)
+			return;
+
+		var progressFinishAction = new FinishProgressAction(
+			reason =>
+			{
+				switch (reason)
+				{
+					case FinishProgressAction.FinishReason.INTERRUPTED:
+						CancelCPR();
+						doingCPR = false;
+						break;
+					case FinishProgressAction.FinishReason.COMPLETED:
+						DoCPR(rescuer, cardiacArrestPlayer);
+						doingCPR = false;
+						break;
+				}
+			}
+		);
+
+		doingCPR = true;
+		UIManager.ProgressBar.StartProgress(cardiacArrestPlayerRegister.WorldPosition, 5f, progressFinishAction,
+			rescuer);
+		ChatRelay.Instance.AddToChatLogServer(new ChatEvent
+		{
+			channels = ChatChannel.Local,
+			message = $"{rescuer.Player()?.Name} is trying to perform CPR on {cardiacArrestPlayer.Player()?.Name}.",
+			position = cardiacArrestPlayerRegister.WorldPosition.To2Int()
+		});
+	}
+
+	[Server]
+	private void DoCPR(GameObject rescuer, GameObject CardiacArrestPlayer)
+	{
+		var CardiacArrestPlayerRegister = CardiacArrestPlayer.GetComponent<RegisterPlayer>();
+		CardiacArrestPlayer.GetComponent<PlayerHealth>().bloodSystem.oxygenDamage -= 7f;
```

Comment:
Idk what are classic values for CPR, but it's really hard to revive a person if you're the only one performing CPR
if it's supposed to be this way, it's fine, otherwise maybe double this value?

Reply:
It is a bit low, I don't think CPR is very good.
In tg's /mob/living/carbon/human/proc/do_cpr(mob/living/carbon/C) it heals 7 oxy damage until oxy damage is 0.
Here is a picture with the healing code highlighted:
![Nz1lupd](https://user-images.githubusercontent.com/28093344/60640611-0d6ea880-9e7c-11e9-98ed-71b764f2a294.png)
Which goes to:
![zxqNqEk](https://user-images.githubusercontent.com/28093344/60640708-8ec63b00-9e7c-11e9-90bd-7898b1762001.png)

Reply:
remember that you can spam cpr on a person and only get the mob clicking cd, you can do several cpr at the same time